### PR TITLE
Size Factor Field for SCVI, MULTIVI, SCANVI, TOTALVI

### LINKED ISSUE #1321: Add `size_factor_key` to relevant models
This would be in `setup_anndata` so that users can put a custom size factor in the likelihood. Setting this should make the output activation of `px_rho` to be a softplus, not a softmax

- [x] scVI
- [x] totalVI
- [x] MultiVI
- [x] scANVI

### PR DESCRIPTION
**Pull request by jjhong922**
Fixes #1321

### CODE REVIEW

File: scvi/nn/_base_components.py [hunk 1]
```diff
@@ -399,7 +412,7 @@ def forward(
         px_scale = self.px_scale_decoder(px)
         px_dropout = self.px_dropout_decoder(px)
         # Clamp to high value: exp(12) ~ 160000 to avoid nans (computational stability)
-        px_rate = torch.exp(library) * px_scale  # torch.clamp( , max=12)
+        px_rate = size_factor * px_scale  # torch.clamp( , max=12)
```
Comment (adamgayoso) [inline scvi/nn/_base_components.py:415]: I know it seems weird, but this is a breaking change now because this class assumed library was on the log scale...

File: scvi/module/_multivae.py [hunk 2]
```diff
@@ -412,8 +424,10 @@ def generative(
         p = self.z_decoder_accessibility(decoder_input, batch_index, *categorical_input)
 
         # Expression Decoder
+        if not self.use_size_factor_key:
+            size_factor = torch.exp(libsize_expr)
```
Comment (adamgayoso) [inline scvi/module/_multivae.py:428]: I would log the size factor instead of exp the lib size (same in scvi vae)

File: scvi/module/_vae.py [hunk 3]
```diff
@@ -322,8 +336,16 @@ def generative(
         if transform_batch is not None:
             batch_index = torch.ones_like(batch_index) * transform_batch
 
+        if not self.use_size_factor_key:
+            size_factor = torch.exp(library)
```
Comment (adamgayoso) [inline scvi/module/_vae.py:340]: log the size factor instead of exp the library (breaking change)

File: scvi/utils/_docstrings.py [hunk 4]
```diff
@@ -75,6 +75,11 @@ class SetupAnnDataDocstringProcessor(DocstringProcessor):
         key in `adata.obs` for label information. Categories will automatically be converted into integer
         categories and saved to `adata.obs['_scvi_labels']`. If `None`, assigns the same label to all the data."""
 
+    param_size_factor_key = """\
+    size_factor_key
+        key in `adata.obs` for size factor information. Instead of using library size as a size factor, the provided
+        size factor column will be used as a coefficient for a softplussed `px_scale` term."""
```
Comment (adamgayoso) [inline scvi/utils/_docstrings.py:81]: "as offset in the mean of the likelihood. Assumed to be on linear scale."

File: tests/models/test_models.py [hunk 5]
```diff
@@ -471,17 +479,22 @@ def test_new_setup_compat():
     model = SCVI(adata)
     model.view_anndata_setup(hide_state_registries=True)
 
+    registry = adata_manager.registry[_constants._FIELD_REGISTRIES_KEY]
```
Comment (adamgayoso) [inline tests/models/test_models.py:482]: what is going on here?

**Review by adamgayoso — CHANGES_REQUESTED** [review]

File: tests/models/test_models.py (hunk 5 above)
Comment (jjhong922) [reply tests/models/test_models.py:482]: oops poor naming. But basically keeping this test working even if new fields are added to SCVI (e.g. size factor) - not supported by backwards compatibility.

**Review by adamgayoso — APPROVED** [review]
